### DISCUSSION TITLE
Directive to skip slide in PDF: `_skip-pdf: true`

### DISCUSSION THREAD

**Discussion by sbibauw** [marp-team/marp#489] [Ideas] [answered]
It would be really useful to have a simple directive available to instruct Marp NOT to print out a specific slide when exporting to PDF.
For instance, I often have to create "transition" slides or step-by-step slides when needing to create an "animation" in an image (in Powerpoint, you'd make elements of a graph appear step-by-step, here I have to generate a new image for every step and create a slide for each. However, when exporting to PDF, I'd like only the last slide to be kept. 
Same thing when I want to add an emphasis on text (e.g., slide 1 "This is an example"; slide 2 "This is an **example**" -> when exporting to PDF, I only want slide 2).

Inserting a slide-specific directive to skip/ignore these "transition" slides when exporting to PDF would be great :

```md
---
# Slide 1 (to be ignored in PDF)
This is an example.
<!-- _skip: pdf -->
---
# Slide 2
This is an **example**.
---
```

or another version of this (`ignore-slide-in-pdf: true`, `hidden-slide: true`, etc.).
`_skip: pdf` or `_hide: pdf` would enable alternatives such as `_skip: all` when a slide is always hidden and `_skip: html` (but I don't see a use case for that).

Would that be possible and easy to implement?

**Comment by yhatt** [comment, marked as answer]
The Marp team does not plan to adopt this proposal at this time because:

- It could allow to make the different contents and semantics from original Markdown and other output formats, and may become easy to lose document consistency.
- Marp already has a lot of directives. We don't want to blindly add features and overwhelm users.

An existing feature of Marp/Marpit that exhibits the behavior you've described is [Fragments](https://marpit.marp.app/fragmented-list) (particularly in the context of PDF output). By using fragments with some CSS tricks, you can achieve the same behavior as your proposal.

```markdown
<style scoped>
@media screen {
  /* Viewing on HTML (Marp CLI's bespoke template): Hide not current fragments */
  [data-marpit-fragment][data-bespoke-marp-fragment]:not([data-bespoke-marp-current-fragment="current"]) {
    display: none;
  }
}

@media print {
  /* Printing to PDF: Hide not last fragments */
  [data-marpit-fragment]:not([data-marpit-fragment]:last-of-type) {
    display: none;
  }
}
</style>

<div data-marpit-fragment>

# 1st fragment

</div>

<div data-marpit-fragment>

# 2nd fragment

</div>

<div data-marpit-fragment>

# 3rd fragment

</div>

---

...
```

This example is defining fragments explicitly by using [`<div data-marpit-fragment>`](https://github.com/marp-team/marpit/issues/195#issuecomment-534819826). The scoped style instructs that it will hide the fragments except the current one when viewing on HTML, and hide the fragments except the last one when printing to PDF.

And using fragments is better habits for maintaining semantic Markdown and HTML rather than using multiple slides. Fragmented contents are not separated into multiple slides. It's just a single slide with multiple fragments. It will not be wrongly recognized as multiple slide pages even if parsed by Markdown processors other than Marp.

Lastly, you also can create Marp plugin to achieve your proposal based on this example, [just like `hide` directive at here](https://github.com/orgs/marp-team/discussions/162#discussioncomment-3415333). When the plugin detected the directive definition, manipulate markdown-it tokens to separate contents into fragments and put specific styles :)
